src/generic/grid.cpp

```diff
@@ -6393,9 +6676,10 @@ void wxGrid::ShowCellEditControl()
 #endif
 
             wxGridCellEditor* editor = attr->GetEditor(this, row, col);
-            if ( !editor->IsCreated() )
+            if ( !editor->IsCreated() || editor->GetParent() != gridWindow)
```

Comment:
It can in the following case:
It could have been created with a gridWindow as a parent, the frozen state changed, and now another one wants to use it.

Reply:
I think it could be simpler to just cancel editing when changing the frozen state.

This would make the logic/state of wxGrid simpler, IMHO.

Reply:
> just cancel editing when changing the frozen state

Interesting, from your comment I've just uncovered a bug (really shouldn't allow edit to exist when changing the frozen state)

But it's a different case here. The editor is created only once, and then stored to be reused.
So it's created with the first gridWindow that needs it, as the parent. Without that check, when another gridWindow calls the editor, it will display incorrectly.